app/src/shared/components/ActivityList/ActivityStreamHandler/index.jsx

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import React, { useEffect, useState } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+
+import { getMyResourceKeys } from '$shared/modules/resourceKey/actions'
+import { selectAuthApiKeyId } from '$shared/modules/resourceKey/selectors'
+import { selectAuthState } from '$shared/modules/user/selectors'
+import { useClientProvider } from '$shared/contexts/StreamrClient'
+import Activity from '$shared/utils/Activity'
+import { isLocalStorageAvailable } from '$shared/utils/storage'
+import { Provider as PendingProvider } from '$shared/contexts/Pending'
+
+const storage = isLocalStorageAvailable() ? localStorage : null
+
+const Handler = () => {
+    const dispatch = useDispatch()
+    const [streamId, setStreamId] = useState(storage ? storage.getItem('user.activityStreamId') : null)
+    const apiKey = useSelector(selectAuthApiKeyId)
+    const authState = useSelector(selectAuthState)
+    const { client } = useClientProvider({
+        apiKey,
+        loadKeys: () => dispatch(getMyResourceKeys()),
+        ...authState,
+    })
+
+    useEffect(() => {
+        const publishActivity = (activity) => {
+            if (client && client.options.auth.apiKey && streamId) {
+                const data = activity.serialize()
+                client.publish(streamId, data)
+            }
+        }
+
+        if (client && apiKey && streamId) {
+            Activity.subscribe(publishActivity)
+        }
+    }, [client, apiKey, streamId])
+
+    useEffect(() => {
+        const createStream = async () => {
+            const stream = await client.createStream({
```

Comment:
Do we want to also remove permissions from this stream?

Reply:
Probably yes. It is still unsure if it is our duty to even create this stream. This could be a backend feature also so let's keep it simple for now until we can enable the feature :)